common/src/stack/command/stack/argument_processors/pallet.py

```diff
@@ -51,3 +96,66 @@ def get_pallets(self, args, params):
 			pallets.extend([Pallet(*row) for row in rows])
 
 		return pallets
+
+	def _normalize_pallet_info(self, pallet_info):
+		"""Normalize the pallet info to a Pallet dataclass."""
+		# If this already has the attributes to be a Pallet, just use it.
+		if all(hasattr(pallet_info, attr_name) for attr_name in PALLET_ATTR_MAPPINGS):
+			return pallet_info
+
+		return Pallet.create_from_probepal_pallet_info(pallet_info)
+
+	def _get_pallet_hooks(self, operation, pallet_info):
+		"""Yield the pallet hooks that should run for this pallet and operation."""
+		# Find all pallet_hooks.json
+		hook_metadata_files = (
+			path for path in PALLET_HOOK_ROOT.glob(f"**/{PALLET_HOOK_METADATA_FILENAME}")
```

Comment:
Why would there be multiple `pallet_hooks.json` files?

Reply:
Because each pallet might have its own that gets added into the root directory under a sub-directory named after the pallet using name, rel, version, os, arch.

Reply:
Would the hooks of one pallet affect the `stack * pallet` operations of another?

Reply:
That's exactly the idea here. We have to patch the base SLES media to get our installer hooks in.

We obviously can't modify and re-distribute the base SLES media. Instead what happens is when the SLES media is enabled for a box, an enable hook installed by the stacki pallet runs and copies over the right files to patch stacki into the base SLES media.

This is also complicated by the fact that once SLES15 lands, it's very possible someone enables the SLES12 base iso before enabling the stacki SLES12 pallet for a given box. In this case the above logic wouldn't be enough to ensure the SLES media is always patched. So the stacki SLES12 pallet also has an enable hook that tries to patch the SLES base media when the stacki SLES12 pallet is enabled, covering both pallet enable ordering scenarios. 

In the end, this is just a facility for running arbitrary scripts during `stack * pallet` operations. When those scripts get run are controlled by the content of the `pallet_hooks.json` file.

Reply:
I see. Thanks your your explanation.